src/crush/CrushWrapper.cc

```diff
@@ -1524,6 +1524,17 @@ int CrushWrapper::get_parent_of_type(int item, int type, int rule) const
   return 0; // not found
 }
 
+void CrushWrapper::get_subtree_of_type(int type, vector<int> *subtrees) const
+{
+  for (int i = 0; i < crush->max_buckets; ++i) {
```

Comment:
this should only do this search from the roots (use get_roots()) or else there is a ton of duplicated work doing recursive descent from every bucket